include/measurement_kit/neubot/negotiate.hpp

```diff
@@ -0,0 +1,22 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef MEASUREMENT_KIT_NEUBOT_NEGOTIATE_HPP
+#define MEASUREMENT_KIT_NEUBOT_NEGOTIATE_HPP
+
+#include <functional>
+#include <measurement_kit/common.hpp>
+
+namespace mk {
+namespace neubot {
+namespace negotiate {
+
+void run_negotiation(Settings settings, Callback<Error> cb,
```

Comment:
I'd rename this function run, because I see lots of sense in it being called as `negotiate::run()`. I'd also make `settings` optional, move it before `reactor` and initialize it with `{}`.